### PR TITLE
Reworking set datastructure

### PR DESCRIPTION
**Pull request by Branchverse**
- Full rework of the backend regarding sets and tasks 
- Removal of most aggregations for better understanding and debugging potential.
- Testing for around 90%

### CODE REVIEW

File: .cli.env [hunk 1]
```diff
@@ -2,7 +2,7 @@
 PORT=3000
 DB_URI=URL_ATLAS
 JWT_SECRET=secretkey
-JWT_EXPIRESIN=10h
+JWT_EXPIRESIN=100h
```
Comment (coffemakingtoaster) [inline .cli.env:5]: u sure about this one? @Branchverse

File: .development.env [hunk 2]
```diff
@@ -11,7 +11,7 @@ DB_URI=mongodb://localhost:27017/TruthOrDare
 
 # JWT
 JWT_SECRET=secretkey
-JWT_EXPIRESIN=10h
+JWT_EXPIRESIN=100h
```
Comment (coffemakingtoaster) [inline .development.env:14]: same thing

File: src/set/entities/set.entity.ts [hunk 3]
```diff
@@ -4,20 +4,20 @@ import { Status } from "../../shared/enums/status.enum";
...
     name: string
 
-    @Prop({ required: false, type: [{ type: TaskSchema}] })
+    @Prop({ required: false, type: [{ type: TaskSchema }] })
     tasks: Task[]
 
     @Prop({ default: Status.ACTIVE })
     status: Status | Status.ACTIVE
 
-    @Prop({ default: "", index: true })
```
Comment (coffemakingtoaster) [inline src/set/entities/set.entity.ts:18]: we removed descriptions? (not a code issue just not entirely sure if this is an oversight or a design choice)

File: src/set/set.controller.ts [hunk 4]
```diff
@@ -19,78 +18,114 @@ export class SetController {
...
+    return this.setService.getAllSets();
+  }
+
+  @Get('/full')
+  @UseGuards(JwtAuthGuard)
+  @ApiOperation({ summary: 'Get all Sets' })
+  getAllSetsFull(
+    @Request() { user }: ParameterDecorator & { user: JwtUserDto }) {
+    throw new NotImplementedException('Currently disabled')
+    return //this.setService.getAllSetsFull(user);
```
Comment (coffemakingtoaster) [inline src/set/set.controller.ts:38]: maybe just remove endpoint for now?

File: src/set/set.controller.ts [hunk 5]
```diff
@@ -19,78 +18,114 @@ export class SetController {
   @Post()
   @UseGuards(JwtAuthGuard)
   @ApiOperation({ summary: 'Create new set' })
-  create(
+  createSet(
     @Body(new ValidationPipe({ whitelist: true })) createSetDto: CreateSetDto, @Request() req) {
```
Comment (coffemakingtoaster) [inline src/set/set.controller.ts:22]: return type?

**Review by coffemakingtoaster — CHANGES_REQUESTED** [review]
unsure if api endpoints should have a return type...other than that lgtm

File: src/set/entities/set.entity.ts (hunk 3 above)
Comment (Branchverse) [reply src/set/entities/set.entity.ts:18]: yes we removed descriptions for the sake of simplicity, the name of a Set should be catchy enough.

**Pull request merged**